x-pack/platform/plugins/shared/spaces/server/plugin.ts

```diff
@@ -155,14 +155,20 @@ export class SpacesPlugin
 
     const { license } = this.spacesLicenseService.setup({ license$: plugins.licensing.license$ });
 
+    let defaultSolution;
+
+    this.config$.pipe(take(1)).subscribe((config) => {
+      defaultSolution = config.defaultSolution;
+    });
+
     this.defaultSpaceService = new DefaultSpaceService();
     this.defaultSpaceService.setup({
       coreStatus: core.status,
       getSavedObjects: async () => (await core.getStartServices())[0].savedObjects,
       license$: plugins.licensing.license$,
       spacesLicense: license,
       logger: this.log,
-      solution: plugins.cloud?.onboarding?.defaultSolution,
+      solution: plugins.cloud?.onboarding?.defaultSolution || defaultSolution,
```

Comment:
Please update the associated test file with a test that covers this new condition